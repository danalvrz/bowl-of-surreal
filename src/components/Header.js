import React from 'react';
import { useNavigate, useLocation } from 'react-router-dom';
import Arrow from '../imgs/arrow.png';
import Dots from '../imgs/dots.png';

const Header = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const { pathname } = location;
  return (
    <div className="w-full fixed z-50">
      <header className="flex sticky h-16 justify-between bg-gray-800 py-4 px-3">
        <button type="button" onClick={() => navigate(-1)}>
          <img
            alt="arrow-back"
            src={Arrow}
            width="25"
            className={`invert ${
              pathname === '/' ? 'opacity-0' : 'opacity-100'
            }`}
          />
        </button>
        <span className="font-Work text-2xl text-white">bowl-of-surreal</span>
        <span className="pr-3">
          <img alt="arrow-back" src={Dots} width="25" className="pt-1 invert" />
        </span>
      </header>
    </div>
  );
};

export default Header;
