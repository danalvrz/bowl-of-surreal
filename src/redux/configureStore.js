// eslint-disable-next-line import/no-extraneous-dependencies
import { createStore, applyMiddleware, compose } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';

const composedEnhancer = compose(applyMiddleware(logger), applyMiddleware(thunk));
const store = createStore(
  rootReducer,
  composedEnhancer,
);

export default store;
