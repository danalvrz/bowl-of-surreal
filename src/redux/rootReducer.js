// eslint-disable-next-line import/no-extraneous-dependencies
import { combineReducers } from 'redux';
import artworks from './artworks/artworks';
import colorsFilter from './colors/colorsFilter';
import responseStatus from './status/status';
import optionsStatus from './options/options';

const rootReducer = combineReducers({
  artworks,
  colorsFilter,
  responseStatus,
  optionsStatus,
});

export default rootReducer;
