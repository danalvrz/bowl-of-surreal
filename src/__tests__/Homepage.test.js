import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Header from '../components/Header';

const mockReducer = () => [];
const store = createStore(mockReducer);

describe('Header section', () => {
  it('renders Header correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
