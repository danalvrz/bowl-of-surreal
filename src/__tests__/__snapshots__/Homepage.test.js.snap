// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header section renders Header correctly 1`] = `
<div
  className="w-full fixed z-50"
>
  <header
    className="flex sticky h-16 justify-between bg-gray-800 py-4 px-3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt="arrow-back"
        className="invert opacity-0"
        src="arrow.png"
        width="25"
      />
    </button>
    <span
      className="font-Work text-2xl text-white"
    >
      bowl-of-surreal
    </span>
    <span
      className="pr-3"
    >
      <img
        alt="arrow-back"
        className="pt-1 invert"
        src="dots.png"
        width="25"
      />
    </span>
  </header>
</div>
`;
